### PR TITLE
[IMP] various: make string translatable

### PR DESCRIPTION
**Pull request by adjo-odoo**
Before this PR, some error messages and others messages are 
not translated.

This PR uses the translation function to be able to translate
those messages.

task-3006627

### CODE REVIEW

**Review by MissingNoShiny — COMMENTED** [review]
LGTM, thanks for your work :slightly_smiling_face:

**Review by xavierbol — COMMENTED** [review]
Thanks for the fix, can you reword the commit message?

```txt
[FIX] {sale_}project,hr_timesheet: translate various messages

Before this commit, some error messages and others messages are 
not translated.

This commit uses the translation function to be able to translate
those messages.

task-3006627
```

Apart of that it looks good to me. 🙂

File: addons/sale_project/models/project.py [hunk 1]
```diff
@@ -566,7 +566,7 @@ def action_project_sharing_view_so(self):
         if not self.display_sale_order_button:
             return {}
         return {
-            "name": "Portal Sale Order",
+            "name": _("Portal Sale Order"),
```
Comment (xavierbol) [inline addons/sale_project/models/project.py:569]: IMO, it is not really useful since this name is not really rendered in the UI. 🤔

Comment (adjo-odoo) [reply addons/sale_project/models/project.py]: Done
Thanks :smile:

Comment (xavierbol) [reply addons/sale_project/models/project.py]: Thanks 🙂

File: addons/project/models/project.py [hunk 2]
```diff
@@ -2491,7 +2491,7 @@ def action_project_sharing_view_parent_task(self):
                 if project._check_project_sharing_access():
                     url = f"/my/projects/{self.parent_id.project_id.id}?task_id={self.parent_id.id}"
                 return {
-                    "name": "Portal Parent Task",
+                    "name": _("Portal Parent Task"),
```
Comment (xavierbol) [inline addons/project/models/project.py:2494]: IMO, it is not really necessary since it is not rendered in the UI. 🤔

File: addons/project/models/project.py [hunk 3]
```diff
@@ -2532,7 +2532,7 @@ def action_project_sharing_open_subtasks(self):
                 action['res_id'] = subtasks.id
             return action
         return {
-            'name': 'Portal Sub-tasks',
+            'name': _('Portal Sub-tasks'),
```
Comment (xavierbol) [inline addons/project/models/project.py:2535]: IMO, it is not really necessary since it is not rendered in the UI. 🤔

File: addons/sale_project/i18n/sale_project.pot [hunk 4]
```diff
@@ -353,6 +353,7 @@ msgid ""
 " the milestone is reached."
 msgstr ""
 
+
```
Comment (xavierbol) [inline addons/sale_project/i18n/sale_project.pot:356]: Unnecessary diff? 🤔

**Review by xavierbol — COMMENTED** [review]
I left 3 last comments and after that it will look good to me. 🙂

File: addons/sale_project/i18n/sale_project.pot (hunk 4 above)
Comment (adjo-odoo) [reply addons/sale_project/i18n/sale_project.pot]: Done
Thanks :smile:

**Review by xavierbol — APPROVED** [review]
LGTM, thanks again for your work. 🙂 

robodoo r+

**Comment by xavierbol** [comment]
robodoo r+

**Comment by xavierbol** [comment]
@robodoo r+
